The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/4525699146/jobs/7970377814) [(plain)](https://github.com/rust-lang/rust/commit/3a2b3a9df34a28202ff892d266b5c4a972b624cd/checks/12285733223/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    |
36  |   macro_rules! assert_eq {
    |   ---------------------- in this expansion of `$crate::assert_eq!` (#2)
...
53  |                   if !(*left_val == *right_val) {
    |                                     ^^^^^^^^^^ expected `(Size, AbiAndPrefAlign)`, found `(Size, Align)`
246 |   macro_rules! debug_assert_eq {
    |   ---------------------------- in this expansion of `debug_assert_eq!` (#1)
...
249 |               $crate::assert_eq!($($arg)*);
249 |               $crate::assert_eq!($($arg)*);
    |               ---------------------------- in this macro invocation (#2)
    |
   ::: compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs:185:13
    |
185 | /             debug_assert_eq!(
186 | |                 (data_layout.pointer_size, data_layout.pointer_align),
187 | |                 cx.size_and_align_of(ptr_type),
188 | |                 "ptr_type={}, pointee_type={}",
189 | |                 ptr_type,
190 | |                 pointee_type,
    | |_____________- in this macro invocation (#1)
    |
    |
    = note: expected tuple `(rustc_target::abi::Size, AbiAndPrefAlign)`
               found tuple `(rustc_target::abi::Size, Align)`

error[E0599]: no method named `bits` found for struct `AbiAndPrefAlign` in the current scope
   --> compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs:198:47
    |
198 |                     data_layout.pointer_align.bits() as u32,
    |                                               ^^^^ method not found in `AbiAndPrefAlign`
help: some of the expressions' fields have a method of the same name
    |
    |
198 |                     data_layout.pointer_align.abi.bits() as u32,
    |                                               ++++
198 |                     data_layout.pointer_align.pref.bits() as u32,

Some errors have detailed explanations: E0308, E0599.
For more information about an error, try `rustc --explain E0308`.
error: could not compile `rustc_codegen_llvm` due to 2 previous errors
```

</details>

--- a/compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs
+++ b/compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs
@@ -176,15 +176,14 @@ fn build_pointer_or_reference_di_node<'ll, 'tcx>(
 
     return_if_di_node_created_in_meantime!(cx, unique_type_id);
 
-    let (thin_pointer_size, thin_pointer_align) =
-        cx.size_and_align_of(cx.tcx.mk_imm_ptr(cx.tcx.types.unit));
+    let data_layout = &cx.tcx.data_layout;
     let ptr_type_debuginfo_name = compute_debuginfo_type_name(cx.tcx, ptr_type, true);
 
     match fat_pointer_kind(cx, pointee_type) {
         None => {
             // This is a thin pointer. Create a regular pointer type and give it the correct name.
             debug_assert_eq!(
-                (thin_pointer_size, thin_pointer_align),
+                (data_layout.pointer_size, data_layout.pointer_align.abi),
                 cx.size_and_align_of(ptr_type),
                 "ptr_type={}, pointee_type={}",
                 ptr_type,
@@ -195,8 +194,8 @@ fn build_pointer_or_reference_di_node<'ll, 'tcx>(
                 llvm::LLVMRustDIBuilderCreatePointerType(
                     DIB(cx),
                     pointee_type_di_node,
-                    thin_pointer_size.bits(),
-                    thin_pointer_align.bits() as u32,
+                    data_layout.pointer_size.bits(),
+                    data_layout.pointer_align.abi.bits() as u32,
                     0, // Ignore DWARF address space.
                     ptr_type_debuginfo_name.as_ptr().cast(),
                     ptr_type_debuginfo_name.len(),

